File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s504715611_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <limits>
using namespace std;

const long long mod = 1000000007LL;
const double eps = 1.0e-14;
const int big = numeric_limits<int>::max();
const double PI = 3.14159265358979323846262338327950288;

class UnionFind {
private:
    vector<int> Parent;
public:
    UnionFind(int n) : Parent(n, -1) {}
    int root(int A) {
        if (Parent[A] < 0) return A;
        return Parent[A] = root(Parent[A]);
    }
    int size(int A) {
        return -Parent[root(A)];
    }
    bool connect(int A, int B) {
        int rootA = root(A), rootB = root(B);
        if (rootA == rootB) return false;
        if (size(rootA) < size(rootB)) swap(rootA, rootB);
        Parent[rootA] += Parent[rootB];
        Parent[rootB] = rootA;
        return true;
    }
};

class FastScanner {
private:
    istream& in;
    vector<char> buffer;
    size_t ptr = 0, buflen = 0;
public:
    FastScanner() : in(cin), buffer(1024) {}
    bool hasNextByte() {
        if (ptr < buflen) return true;
        ptr = 0;
        buflen = in.read(buffer.data(), buffer.size());
        return buflen > 0;
    }
    int readByte() {
        if (!hasNextByte()) return -1;
        return static_cast<unsigned char>(buffer[ptr++]);
    }
    bool hasNext() {
        while (hasNextByte()) {
            int current = buffer[ptr];
            if (current >= 33 && current <= 126) return true;
            ptr++;
        }
        return false;
    }
    string next() {
        if (!hasNext()) throw runtime_error("No element");
        string sb;
        int b = readByte();
        while (b >= 33 && b <= 126) {
            sb += static_cast<char>(b);
            b = readByte();
        }
        return sb;
    }
    long long nextLong() {
        if (!hasNext()) throw runtime_error("No element");
        long long n = 0;
        bool minus = false;
        int b = readByte();
        if (b == '-') {
            minus = true;
            b = readByte();
        }
        if (b < '0' || b > '9') throw invalid_argument("Invalid number format");
        while (true) {
            if (b >= '0' && b <= '9') {
                n = n * 10 + (b - '0');
            } else if (b == -1 || !isPrintableChar(b)) {
                return minus ? -n : n;
            } else {
                throw invalid_argument("Invalid number format");
            }
            b = readByte();
        }
    }
    int nextInt() {
        long long nl = nextLong();
        if (nl < INT_MIN || nl > INT_MAX) throw out_of_range("Value out of range");
        return static_cast<int>(nl);
    }
    double nextDouble() {
        string s = next();
        return stod(s);
    }
    char nextchar() {
        if (!hasNextByte()) throw runtime_error("No more input");
        return static_cast<char>(readByte());
    }
private:
    static bool isPrintableChar(int c) {
        return 33 <= c && c <= 126;
    }
};

FastScanner scan;

long long modlcm(long long a, long long b) {
    return a * b * modint(gcd(a, b), mod);
}

long long gcd(long long a, long long b) {
    return b > 0 ? gcd(b, a % b) : a;
}

long long lcm(long long a, long long b) {
    return a / gcd(a, b) * b;
}

int upper_division(int a, int b) {
    return (a + b - 1) / b;
}

long long lupper_division(long long a, long long b) {
    return (a + b - 1) / b;
}

vector<int> setArray(int a) {
    vector<int> b(a);
    for (int i = 0; i < a; ++i) b[i] = scan.nextInt();
    return b;
}

vector<long long> lsetArray(int a) {
    vector<long long> b(a);
    for (int i = 0; i < a; ++i) b[i] = scan.nextLong();
    return b;
}

string reverse(const string& s) {
    string t = s;
    reverse(t.begin(), t.end());
    return t;
}

void printArray(const vector<int>& que) {
    for (size_t i = 0; i < que.size() - 1; ++i) cout << que[i] << " ";
    cout << que.back() << endl;
}

vector<vector<int>> doublesort(const vector<vector<int>>& a) {
    vector<vector<int>> sorted = a;
    sort(sorted.begin(), sorted.end(), [](const auto& x, const auto& y) {
        return x[0] < y[0];
    });
    return sorted;
}

long long modpow(long long x, long long n, long long mo) {
    long long sum = 1;
    while (n > 0) {
        if (n & 1) sum = sum * x % mo;
        x = x * x % mo;
        n >>= 1;
    }
    return sum;
}

vector<char> revch(const vector<char>& ch) {
    vector<char> ret(ch.size());
    for (int i = ch.size() - 1, j = 0; i >= 0; --i, ++j) ret[j] = ch[i];
    return ret;
}

vector<int> revint(const vector<int>& ch) {
    vector<int> ret(ch.size());
    for (int i = ch.size() - 1, j = 0; i >= 0; --i, ++j) ret[j] = ch[i];
    return ret;
}

void warshall_floyd(vector<vector<int>>& v, int n) {
    for (int k = 0; k < n; ++k)
        for (int i = 0; i < n; ++i)
            for (int j = 0; j < n; ++j)
                v[i][j] = min(v[i][j], v[i][k] + v[k][j]);
}

long long modint(long long a, long long m) {
    long long b = m, u = 1, v = 0;
    while (b != 0) {
        long long t = a / b;
        a -= t * b;
        swap(a, b);
        u -= t * v;
        swap(u, v);
    }
    u %= m;
    return u < 0 ? u + m : u;
}

int main() {
    string s = scan.next();
    cout << (s.length() >= 6 && s[2] == s[3] && s[4] == s[5] ? "Yes" : "No") << endl;
    return 0;
} // End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s504715611_commented.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"sort"
	"strconv"
)

type UnionFind struct {
	Parent []int
}

func NewUnionFind(n int) *UnionFind {
	uf := &UnionFind{
		Parent: make([]int, n),
	}
	for i := range uf.Parent {
		uf.Parent[i] = -1
	}
	return uf
}

func (uf *UnionFind) root(A int) int {
	if uf.Parent[A] < 0 {
		return A
	}
	uf.Parent[A] = uf.root(uf.Parent[A])
	return uf.Parent[A]
}

func (uf *UnionFind) size(A int) int {
	return -uf.Parent[uf.root(A)]
}

func (uf *UnionFind) connect(A, B int) bool {
	rootA := uf.root(A)
	rootB := uf.root(B)
	if rootA == rootB {
		return false
	}
	if uf.size(A) < uf.size(B) {
		A, B = B, A
		rootA, rootB = rootB, rootA
	}
	uf.Parent[rootA] += uf.Parent[rootB]
	uf.Parent[rootB] = rootA
	return true
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func lmax(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

func lmin(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func factorial(i int) int {
	if i == 1 {
		return 1
	}
	return i * factorial(i-1)
}

func isPrime(n int64) bool {
	if n == 2 {
		return true
	}
	if n < 2 || n%2 == 0 {
		return false
	}
	sqrt := math.Sqrt(float64(n))
	for i := int64(3); i <= int64(sqrt); i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func upper_division(a, b int) int {
	if a%b == 0 {
		return a / b
	}
	return a/b + 1
}

func lupper_division(a, b int64) int64 {
	if a%b == 0 {
		return a / b
	}
	return a/b + 1
}

func lower_bound(a []int, key int) int {
	left, right := 0, len(a)
	for right-left > 1 {
		mid := (right + left) / 2
		if a[mid] < key {
			left = mid
		} else {
			right = mid
		}
	}
	return left
}

func upper_bound(a []int, key int) int {
	left, right := 0, len(a)
	for right-left > 1 {
		mid := (right + left) / 2
		if a[mid] <= key {
			left = mid
		} else {
			right = mid
		}
	}
	return left
}

func modpow(x, n, mo int64) int64 {
	sum := int64(1)
	for n > 0 {
		if (n & 1) == 1 {
			sum = sum * x % mo
		}
		x = x * x % mo
		n >>= 1
	}
	return sum
}

func modint(a, m int64) int64 {
	b := m
	var u, v int64 = 1, 0
	for b != 0 {
		t := a / b
		a -= t * b
		x := a
		a = b
		b = x
		u -= t * v
		x = u
		u = v
		v = x
	}
	u %= m
	if u < 0 {
		u += m
	}
	return u
}

func setArray(a int, sc *bufio.Scanner) []int {
	res := make([]int, a)
	for i := 0; i < a; i++ {
		sc.Scan()
		num, _ := strconv.Atoi(sc.Text())
		res[i] = num
	}
	return res
}

func lsetArray(a int, sc *bufio.Scanner) []int64 {
	res := make([]int64, a)
	for i := 0; i < a; i++ {
		sc.Scan()
		num, _ := strconv.ParseInt(sc.Text(), 10, 64)
		res[i] = num
	}
	return res
}

func reverse(s string) string {
	r := []rune(s)
	for i, j := 0, len(r)-1; i < j; i, j = i+1, j-1 {
		r[i], r[j] = r[j], r[i]
	}
	return string(r)
}

func revch(ch []rune) string {
	reversed := make([]rune, len(ch))
	for i, j := 0, len(ch)-1; i < len(ch); i, j = i+1, j-1 {
		reversed[i] = ch[j]
	}
	return string(reversed)
}

func revint(arr []int) []int {
	reversed := make([]int, len(arr))
	for i, j := 0, len(arr)-1; i < len(arr); i, j = i+1, j-1 {
		reversed[i] = arr[j]
	}
	return reversed
}

func warshall_floyd(v [][]int, n int) {
	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				if v[i][j] > v[i][k]+v[k][j] {
					v[i][j] = v[i][k] + v[k][j]
				}
			}
		}
	}
}

func doublesort(a [][]int) [][]int {
	sort.Slice(a, func(i, j int) bool { return a[i][0] < a[j][0] })
	return a
}

func ldoublesort(a [][]int64) [][]int64 {
	sort.Slice(a, func(i, j int) bool { return a[i][0] < a[j][0] })
	return a
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	str := scanner.Text()
	if len(str) < 6 {
		fmt.Println("No")
		return
	}
	if str[2] == str[3] && str[4] == str[5] {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

// End of Code
